ethereum/core/src/test/java/org/hyperledger/besu/ethereum/mainnet/EtcHashTest.java

```diff
@@ -14,21 +14,19 @@
  */
 package org.hyperledger.besu.ethereum.mainnet;
 
-import java.util.function.Function;
-
 import org.assertj.core.api.Assertions;
 import org.junit.Test;
 
 public class EtcHashTest {
 
   @Test
   public void testEpoch() {
-    Function<Long, Long> epochCalculator = EthHash.ecip1099Epoch(2_000_000);
+    EpochCalculator epochCalculator = new EpochCalculator.Ecip1099EpochCalculator(2_000_000);
 
     // check before activation block (1,000,000/30,000 = 33)
-    Assertions.assertThat(epochCalculator.apply(1_000_000L)).isEqualTo(33);
```

Comment:
(a) rather than testing a before/after activation block (which per another comment I think we should get rid of the activation block) we should test early (somewhere in the first 30k blocks) vs late (somewhere in the last 30k blocks)

(b) we don't test `epochStartBlock`, this seems like a reasonable place to add that as a unit test.

Reply:
Tests added.